### PR TITLE
Avalara credentials, channel configuration, tax category mapping

### PR DESCRIPTION
**Pull request by katienesterovich**
Hi Riccardo! Please rebase branch to liferay-portal/master 198c693cbd9c prior to reviewing.

### [COMMERCE-5296](https://issues.liferay.com/browse/COMMERCE-5296) - Group UI inputs for Avalara credentials/settings 
 - Re-grouped UI to be more user-friendly. Updated again in COMMERCE-5371, see below
- If Global credentials invalid: 
image
- If Global credentials valid: 
image

### [COMMERCE-5297](https://issues.liferay.com/browse/COMMERCE-5297) - Make Company Code input a dropdown with avalara values
- Company codes are now displayed in a dropdown (Company name is displayed, but company Code is registered)
- If not specified or get call failed, will be set to DEFAULT
   image

### [COMMERCE-5371](https://issues.liferay.com/browse/COMMERCE-5371) - Move credentials configuration to a global scope

- Application Menu (Under Settings > Avalara)

image

- Currently one tab - credentials

image

- Failed Verify Connection 

image

- Successful verification

image

### [COMMERCE-4348](https://issues.liferay.com/browse/COMMERCE-4348) - Add a health check to map Tax Categories to Avalara TaxCodes

- Health check

image

- Tax Categories created 

image

### QUESTION: Multiple instances of "Expired Tax Code - Do Not Use" Should we skip syncing these?
image

### CODE REVIEW

File: modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java [hunk 1]
```diff
@@ -47,6 +49,18 @@
 )
 public class CommerceAvalaraConnectorImpl implements CommerceAvalaraConnector {
 
+	@Override
+	public List<TaxCodeModel> listTaxCodes() throws Exception {
+		AvaTaxClient avaTaxClient = _getAvaTaxClient();
+
+		// Check if need to paginate, docs say 1000 results max per call
```
Comment (katienesterovich) [inline modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java:56]: Need to remove comment. After running the job, 4,937 taxCategories were created.

File: modules/dxp/apps/commerce-avalara/commerce-avalara-connector-api/src/main/java/com/liferay/commerce/avalara/connector/configuration/CommerceAvalaraConnectorConfiguration.java [hunk 2]
```diff
@@ -16,18 +16,13 @@
 
 import aQute.bnd.annotation.metatype.Meta;
 
-import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
-
 /**
- * @author Calvin Keum
+ * @author Katie Nesterovich
  */
-@ExtendedObjectClassDefinition(
```
Comment (riccardo-alberti) [inline modules/dxp/apps/commerce-avalara/commerce-avalara-connector-api/src/main/java/com/liferay/commerce/avalara/connector/configuration/CommerceAvalaraConnectorConfiguration.java:24]: Here i think we need to add the annotation for having system scope
ExtendedObjectClassDefinition.Scope.SYSTEM

File: modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java (hunk 1 above)
Comment (riccardo-alberti) [reply modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java:56]: Is this still an open point?

File: modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java [hunk 3]
```diff
@@ -54,6 +84,21 @@ public void verifyConnection(
 		}
 	}
 
+	private AvaTaxClient _getAvaTaxClient() throws PortalException {
+		CommerceAvalaraConnectorConfiguration
```
Comment (riccardo-alberti) [inline modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java:102]: Have you tested with multiple companies to see if everything works as expected?

File: modules/dxp/apps/commerce-avalara/commerce-avalara-tax-engine-fixed-web/src/main/java/com/liferay/commerce/avalara/tax/engine/fixed/web/internal/servlet/taglib/ui/CommerceTaxMethodAvalaraScreenNavigationCategory.java [hunk 4]
```diff
@@ -137,6 +165,38 @@ public void render(
...
+		catch (Exception exception) {
+		}
+
+		httpServletRequest.setAttribute("avalaraCompanies", companies);
+	}
+
+	private boolean _verifyConnection(
+		CommerceAvalaraConnectorConfiguration
+			commerceAvalaraConnectorConfiguration) {
+
```
Comment (riccardo-alberti) [inline modules/dxp/apps/commerce-avalara/commerce-avalara-tax-engine-fixed-web/src/main/java/com/liferay/commerce/avalara/tax/engine/fixed/web/internal/servlet/taglib/ui/CommerceTaxMethodAvalaraScreenNavigationCategory.java:183]: I think it's more readable to pass the companyId to _verifyConnection and get the configuration from inside the _verify

File: modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java (hunk 3 above)
Comment (katienesterovich) [reply modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java:102]: Yes, tested with 3

File: modules/dxp/apps/commerce-avalara/commerce-avalara-connector-api/src/main/java/com/liferay/commerce/avalara/connector/configuration/CommerceAvalaraConnectorConfiguration.java (hunk 2 above)
Comment (katienesterovich) [reply modules/dxp/apps/commerce-avalara/commerce-avalara-connector-api/src/main/java/com/liferay/commerce/avalara/connector/configuration/CommerceAvalaraConnectorConfiguration.java:24]: Hmm, if I add the annotation, then it appears in the System Configuration (currently we have that happen with `@ExtendedObjectClassDefinition(
	category = "tax-rate", scope = ExtendedObjectClassDefinition.Scope.GROUP
)`

And it's displayed in both system and instance settings. 
image

image

image

Do we want that? 
If yes, should add it. If no, should remove from the channel configuration too.

File: modules/dxp/apps/commerce-avalara/commerce-avalara-tax-engine-fixed-web/src/main/java/com/liferay/commerce/avalara/tax/engine/fixed/web/internal/servlet/taglib/ui/CommerceTaxMethodAvalaraScreenNavigationCategory.java (hunk 4 above)
Comment (katienesterovich) [reply modules/dxp/apps/commerce-avalara/commerce-avalara-tax-engine-fixed-web/src/main/java/com/liferay/commerce/avalara/tax/engine/fixed/web/internal/servlet/taglib/ui/CommerceTaxMethodAvalaraScreenNavigationCategory.java:173]: Agreed, very nice find! 😄

File: modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java (hunk 1 above)
Comment (katienesterovich) [reply modules/dxp/apps/commerce-avalara/commerce-avalara-connector/src/main/java/com/liferay/commerce/avalara/connector/internal/CommerceAvalaraConnectorImpl.java]: Removed comment - after running the job, 4,937 taxCategories were created so this is not an issue it seems.

**Pull request closed without merging**